psamm/datasource/native.py

```diff
@@ -307,14 +307,26 @@ def context(self):
         return self._context
 
 
+def checkid(checking):
+    """Check whether the ID is valid.
+
+    Does so by checking if type is a string or length is 0."""
+
+    if checking is None:
+        raise ParseError('Compound ID missing')
+    elif type(checking) is not str:
```

Comment:
What happens if the ID has unicode characters? You may need to use `isinstance` instead of `type` to get around this ( see https://stackoverflow.com/questions/1549801/differences-between-isinstance-and-type-in-python)